vscode-extension/editor/src/VSCodeEditor.tsx

```diff
@@ -0,0 +1,449 @@
+  const setupTelemetryIfAllowed = useCallback(async () => {
+    const isDev = (process.env.NODE_ENV ?? "development") === "development";
+
+    // Don't enable telemetry in dev mode because hot reload will spam the logs.
+    if (isDev) {
+      return;
+    }
+
+    const res = await ufetch.get(
+      ROUTE_TABLE.GET_AICONFIGRC(aiConfigServerUrl),
+      {}
+    );
+
+    const enableTelemetry = res.allow_usage_data_sharing;
+
+    if (enableTelemetry) {
+      datadogLogs.init({
+        clientToken: "pub356987caf022337989e492681d1944a8",
+        env: process.env.NODE_ENV ?? "development",
+        service: "aiconfig-editor",
+        site: "us5.datadoghq.com",
+        forwardErrorsToLogs: true,
+        sessionSampleRate: 100,
+      });
+
+      datadogLogs.setGlobalContextProperty("mode", MODE);
+    }
+  }, [aiConfigServerUrl]);
+
+  useEffect(() => {
+    setupTelemetryIfAllowed();
+  }, [setupTelemetryIfAllowed]);
+
+  const getModels = useCallback(
+    async (search: string) => {
+      // For now, rely on caching and handle client-side search filtering
+      // We will use server-side search filtering for Gradio
+      const res = await ufetch.get(ROUTE_TABLE.LIST_MODELS(aiConfigServerUrl));
+      const models = res.data;
+      if (search && search.length > 0) {
+        const lowerCaseSearch = search.toLowerCase();
+        return models.filter(
+          (model: string) =>
+            model.toLocaleLowerCase().indexOf(lowerCaseSearch) >= 0
+        );
+      }
+      return models;
+    },
+    [aiConfigServerUrl]
+  );
+
+  const addPrompt = useCallback(
+    async (promptName: string, promptData: Prompt, index: number) => {
+      const res = await ufetch.post(ROUTE_TABLE.ADD_PROMPT(aiConfigServerUrl), {
+        prompt_name: promptName,
+        prompt_data: promptData,
+        index,
+      });
+
+      if (vscode) {
+        notifyDocumentDirty(vscode);
+      }
+
+      return res;
+    },
+    [aiConfigServerUrl, vscode]
+  );
+
+  const deletePrompt = useCallback(
+    async (promptName: string) => {
+      const res = await ufetch.post(
+        ROUTE_TABLE.DELETE_PROMPT(aiConfigServerUrl),
+        {
+          prompt_name: promptName,
+        }
+      );
+
+      if (vscode) {
+        notifyDocumentDirty(vscode);
+      }
+
+      return res;
+    },
+    [aiConfigServerUrl, vscode]
+  );
+
+  const clearOutputs = useCallback(async () => {
+    const res = await ufetch.post(
+      ROUTE_TABLE.CLEAR_OUTPUTS(aiConfigServerUrl),
+      {}
+    );
+
+    if (vscode) {
+      notifyDocumentDirty(vscode);
+    }
+
+    return res;
+  }, [aiConfigServerUrl, vscode]);
+
+  const runPrompt = useCallback(
+    async (
+      promptName: string,
+      onStream: RunPromptStreamCallback,
+      onError: RunPromptStreamErrorCallback,
+      enableStreaming: boolean = true,
+      cancellationToken?: string
+    ) => {
+      if (vscode) {
+        notifyDocumentDirty(vscode);
+      }
```

Comment:
Is this needed here? I see it done later on in this function as well. Seems like it should only be done later on to be consistent with other places (done after the request is sent)

Reply:
Oh maybe to handle the streaming, it's done before instead of on stream events